src/Behat/Behat/Context/Snippet/Generator/ContextSnippetGenerator.php

```diff
@@ -122,7 +123,7 @@ public function generateSnippet(Environment $environment, StepNode $step): Snipp
      */
     private function getMethodName(string $contextClass, string $canonicalText, string $pattern): string
     {
-        $methodName = $this->deduceMethodName($canonicalText);
+        $methodName = SimpleStepMethodNameGenerator::cleanupMethodName($canonicalText);
```

Comment:
should we trigger a deprecation here when the cleaned value is different from the provided argument, to ask using the StepMethodNameGenerator in the pattern policy ?

Reply:
The problem is we have no neat way to report that deprecation to the user at the moment, and even if we did if they are consuming a PatternPolicy from an extension that could be difficult for them to resolve in the short term.

Considering all the method actually does is lowercase the first letter and backfill empty strings, I almost considered removing it now - worst case someone will get a suggested method with an uppercase first character (or perhaps no name) and have to fix it in their Context class... It's not going to actually break anything. WDYT?

Reply:
@carlos-granados I have reflected more on @stof's comment I think the benefit of lowercasing the first character (in the unlikely situation someone is using a custom PatternPolicy) is limited for the cost of exposing this method and changing it again at some future date. I've reworked to just handle the case where the method name is empty, which avoids having to have this method at all.